zip-0213.rst

```diff
@@ -0,0 +1,179 @@
+RFC 2119. [#RFC2119]_
+
+The term "network upgrade" in this document is to be interpreted as described in ZIP 200
+[#zip-0200]_.
+
+The term "Sapling" in this document is to be interpreted as described in ZIP 205
+[#zip-0205]_.
+
+The terms "Founders' Reward", "funding stream", and "active funding stream" in this
+document are to be interpreted as described in ZIP 207 [#zip-0207]_.
+
+Abstract
+========
+
+This proposal defines modifications to the Zcash consensus rules that enable coinbase
+funds to be mined to Sapling addresses. It does not disable the use of transparent
+addresses in coinbase transactions.
+
+
+Motivation
+==========
+
+Zcash inherited the concept of "coinbase transactions" from Bitcoin: special transactions
+inside each block that are allowed to have no inputs. These transactions are created by
+miners during block creation, and collect the block reward and transaction fees into new
+transparent outputs that can then be spent. They are also leveraged in Zcash for the
+Founders' Reward (and potentially for funding streams [#zip-0207]_).
+
+On the path to deprecating and removing Bitcoin-inherited transparent addresses within the
+Zcash network, a required step is to be able to create coinbase transactions that have no
+transparent outputs. However, Zcash was launched with a consensus rule preventing coinbase
+transactions from containing shielded outputs, instead enforcing that coinbase funds could
+not be spent in transactions with transparent outputs. This was partly in order to reduce
+the complexity of the original Zcash modifications to the Bitcoin Core codebase, but also
+because at the time, shielded transactions required significant memory and CPU resources
+to create.
+
+The Sapling network upgrade [#zip-0205]_ deployed architectural changes and performance
+improvements that make shielding funds directly in the coinbase transaction feasible. In
+order to reduce the complexity of the Sapling network upgrade, the existing consensus
+rules preventing coinbase transactions from containing shielded outputs were extended to
+cover Sapling outputs. Therefore, it is now necessary to modify the consensus rules in
+order to enable miners to start using Sapling addresses.
+
+
+Specification
+=============
+
+Prior to activation of the TODO network upgrade, the existing consensus rule requiring
+coinbase transactions to have zero Sapling outputs is enforced.
+
+Once the TODO network upgrade activates:
+
+- The above existing consensus rule is no longer active, and coinbase transactions MAY
+  contain Sapling outputs.
+
+- The consensus rules applied to ``valueBalance``, ``vShieldedOutput``, and ``bindingSig``
+  in non-coinbase transactions MUST also be applied to coinbase transactions.
+
+- The existing consensus rule requiring transactions that spend coinbase outputs to have
+  an empty ``vout``, is amended to only apply to transactions that spend transparent
```

Comment:
I don't quite get the rational here. Is this really saying shielded coinbase outputs can be sent directly to transparent addresses?

Reply:
No. What this line is doing is narrowing an existing consensus rule, because after this ZIP is deployed, the existing consensus rule is unenforceable as written. This consensus rule cannot be enforced on transactions that "spend coinbase outputs" (now that this includes shielded outputs), because transactions spending shielded coinbase outputs cannot be distinguished from transactions spending other shielded outputs. Thus the rule must be narrowed to only cover transactions that "spend transparent coinbase outputs".